### PR TITLE
Network switching ui

### PR DESCRIPTION
**Pull request by marijnschilling**
develop

This PR contains:
- [x] refactor of the cell selection in SettingsController
- [x] Add Develop cell to SettingsController
- [x] Remove App version cell from Settingscontroller

- [x] Add develop controller which will contain more developer settings in the future, which shows version and build number
- [x] Network controller for network selection

- [x] Attached to network switching persistancy

### CODE REVIEW

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+class NetworkSettingsController: UITableViewController {
+
+    enum Network: Int {
+        case ropstenRevivalTestnetStaging
+        case ropstenRevivalTestnet
+    }
+
+    @IBOutlet var networkCells: [UITableViewCell]!
+
+    var network: Network?
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: fileprivate ?

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import SweetUIKit
+
+class NetworkSettingsController: UITableViewController {
+
+    enum Network: Int {
+        case ropstenRevivalTestnetStaging
+        case ropstenRevivalTestnet
+    }
+
+    @IBOutlet var networkCells: [UITableViewCell]!
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: fileprivate ?

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    var network: Network?
+
+    var selectedIndexPath: IndexPath? {
+        didSet {
+            guard let selectedIndexPath = selectedIndexPath else { return }
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: is there a case when we dnt have selectedIndexPath (selected network) ?

Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: i see no sense in this guard :(

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //TODO: get current network value from API
+        let currentNetwork = Network.ropstenRevivalTestnetStaging
+
+        selectedIndexPath = IndexPath(row: currentNetwork.rawValue, section: 0)
+        updateCells()
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: please, call this once, on `selectedIndexPath` didSet

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+             let isCurrentNetwork = indexPath == selectedIndexPath
+
+            networkCell.accessoryType = isCurrentNetwork ? .checkmark : .none
+         }
+    }
+}
+
+extension NetworkSettingsController {
+    public override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let cell = tableView.cellForRow(at: indexPath) else { return }
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: `tableView.deselectRow(at indexPath)...`

File: Toshi/Controllers/Settings/SettingsController.swift [hunk 6]
```diff
@@ -205,66 +223,78 @@ open class SettingsController: UITableViewController {
...
-            case 2:
-                self.handleSignOut()
-            default:
-                break
-            }
+        switch profileCellType {
+        case .user:
+            self.navigationController?.pushViewController(ProfileController(), animated: true)
+        case .qrCode:
+            guard let current = TokenUser.current else { return }
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/SettingsController.swift]: can we have some handy method to push controller for those 2 params and call it in each case ?

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift (hunk 2 above)
Comment (marijnschilling) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: Can @IBOutlets be fileprivate?

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift (hunk 3 above)
Comment (marijnschilling) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: yes, on init.

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift (hunk 1 above)
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: yep

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift (hunk 3 above)
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: which one is default ? just assign by default some of them. Can't be undefined state

File: Toshi/Controllers/Settings/SettingsController.swift (hunk 6 above)
Comment (marijnschilling) [inline Toshi/Controllers/Settings/SettingsController.swift]: Hmmm but in the one case i need to push the `AddMoneyController` and in the other the `QRCodeController` Or would you suggest making an enum to represent these two and switch over them?

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 7]
```diff
@@ -30,23 +30,23 @@ class NetworkSettingsController: UITableViewController {
     fileprivate var selectedIndexPath: IndexPath = IndexPath(row: Network.ropstenRevivalTestnetStaging.rawValue, section: 0) {
         didSet {
             network = Network(rawValue: selectedIndexPath.row) ?? Network.ropstenRevivalTestnetStaging
+            updateCells()
         }
     }
 
-    open override func viewDidLoad() {
-        super.viewDidLoad()
+    open override func viewDidAppear(_ animated: Bool) {
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: why didAppear? can't be on willAppear?

Comment (marijnschilling) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: No :( Then `let indexPath = tableView.indexPath(for: networkCell)` in `updateCells()` returns nil. I want that but can't 😭

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        return NetworkSwitcher.shared.availableNetworks.count
+    }
+}
+
+extension NetworkSettingsController: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let network = NetworkSwitcher.shared.availableNetworks[indexPath.row]
+
+        self.showActivityIndicator()
```
Comment (marijnschilling) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: you can guard return when the current active network is selected. Now the spinner will just keep spinning, spinning, spinning 🎶

File: Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        view.tableFooterView = UIView()
+
+        view.register(UITableViewCell.self)
+
+        return view
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
```
Comment (marijnschilling) [inline Toshi/Controllers/Settings/Develop/Network/NetworkSettingsController.swift]: The title of the ViewController should be set to "Network"

**Pull request merged**